src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -44,4 +43,6 @@ public void setData(Model model) {
     protected void indicateAttemptToExecuteIncorrectCommand() {
         EventsCenter.getInstance().post(new IncorrectCommandAttemptedEvent(this));
     }
+    
+    public abstract boolean modifiesData();
```

Comment:
>  change attribute to an abstract mathod, so when new commands are extended from Command eclipse will remind developer to state whether the command modifies data

One of the big problems with this approach is that is opens up the possibility that a command may declare that it does not modify data, but due to sloppy programming, it actually does. Is there a less intrusive way to do this that does not involve making all Commands implement an extra method?

Hmm, stepping back a bit, we know that whenever a Command modifies model data, it must call the relevant method of the model. e.g., `Model::setEventTask()`, `Model::removeDeadlineTask()` etc. Aha, perhaps `ModelManager` may be a better place to keep track of whether commands are modifying data or not?